src/Microsoft.Framework.PackageManager/Publish/PublishManager.cs

```diff
@@ -112,7 +113,19 @@ public bool Publish()
 
             foreach (var runtime in _options.Runtimes)
             {
-                var frameworkName = DependencyContext.GetFrameworkNameForRuntime(Path.GetFileName(runtime));
+                var frameworkName = DependencyContext.SelectFrameworkNameForRuntime(
+                    project.GetTargetFrameworks().Select(x => x.FrameworkName), 
+                    Path.GetFileName(runtime));
```

Comment:
Can we fix a small bug here BTW? `runtime` can be a path to a runtime folder like
```
dnu publish --runtime C:\Users\wwa\.dnx\runtimes\dnx-clr-win-x86.1.0.0-beta6-12208\
```
Pay attention to the trailing slash : )